packages/playwright-core/src/remote/playwrightServer.ts

```diff
@@ -94,38 +96,171 @@ export class PlaywrightServer {
         } catch (e) {
         }
 
-        // Instantiate playwright for the extension modes.
         const isExtension = this._options.mode === 'extension';
-        let clientType: ClientType = 'launch-browser';
-        let semaphore: Semaphore = browserSemaphore;
-        if (isExtension && url.searchParams.has('debug-controller')) {
-          clientType = 'controller';
-          semaphore = controllerSemaphore;
-        } else if (isExtension) {
-          clientType = 'reuse-browser';
-          semaphore = reuseBrowserSemaphore;
-        } else if (this._options.mode === 'launchServer' || this._options.mode === 'launchServerShared') {
-          clientType = 'pre-launched-browser-or-android';
-          semaphore = browserSemaphore;
+        const allowFSPaths = isExtension;
+        launchOptions = filterLaunchOptions(launchOptions, allowFSPaths);
+
+        if (isExtension) {
+          if (url.searchParams.has('debug-controller')) {
+            return new PlaywrightConnection(
+                controllerSemaphore,
+                ws,
+                true,
+                this._playwright,
+                async () => { throw new Error('shouldnt be used'); },
+                id,
+            );
+          }
+          return new PlaywrightConnection(
+              reuseBrowserSemaphore,
+              ws,
+              false,
+              this._playwright,
+              async () => {
```

Comment:
Let's turn these arrow functions into named methods as they were before.

Reply:
I did that in https://github.com/microsoft/playwright/pull/36369/commits/375d1f942d240c3bd6cfa5592551e37d18e116b3, but I don't like it. It adds indirection, but doesn't help readability.